cpp/Cipher/MGLGenerateKeyPairInstaller.cpp

```diff
@@ -29,44 +32,78 @@ namespace margelo {
 
 std::mutex m;
 
-// Current implementation only supports RSA schemes (check line config.variant =
-// ) As more encryption schemes are added this will require an abstraction that
-// supports more schemes
 FieldDefinition getGenerateKeyPairFieldDefinition(
     std::shared_ptr<react::CallInvoker> jsCallInvoker,
     std::shared_ptr<DispatchQueue::dispatch_queue> workerQueue) {
   return buildPair(
       "generateKeyPair", JSIF([=]) {
-        auto config = std::make_shared<RsaKeyPairGenConfig>(
+
+        KeyVariant variant =
+          static_cast<KeyVariant>((int)arguments[0].asNumber());
```

Comment:
There's no real validation here, I guess it's fine but bad things will happen if the user doesn't pass a valid number.

For such enums (enum -> number and number -> enum), I usually always build switch statements. This is a bit of extra code, but it ensures that 
1. The numbers/nominals are always correct (let's say someone adds a value in the enum in between other values, then their nominal will change in value and the cast will fail in the future - without a compiler error!)
2. There are no unknowns, and the user cannot pass out of bounds (with nice error messages instead of just weird fails or even worst case crashes)

I guess it's fine here but again usually switches are better. They are not faster, but safer.